### PR TITLE
Move EvolvedOperatorAnsatz to NLocal directory

### LINKED ISSUE #8326: Move the EvolvedOperatorAnsatz to N_local circuits directory
### Environment

- **Qiskit Terra version**: 0.21.0
- **Python version**: all
- **Operating system**: all

### What is happening?

This issue is regarding the organization of the circuit library, and in particular of the N-local circuits:
https://qiskit.org/documentation/apidoc/circuit_library.html#n-local-circuits

[EvolvedOperatorAnsatz](https://qiskit.org/documentation/stubs/qiskit.circuit.library.EvolvedOperatorAnsatz.html#qiskit.circuit.library.EvolvedOperatorAnsatz) gate is not under N-local circuits, although its basis is [NLocal](https://qiskit.org/documentation/stubs/qiskit.circuit.library.NLocal.html#qiskit.circuit.library.NLocal),
While [QAOAAnsatz](https://qiskit.org/documentation/stubs/qiskit.circuit.library.QAOAAnsatz.html#qiskit.circuit.library.QAOAAnsatz) gate whose basis is [EvolvedOperatorAnsatz](https://qiskit.org/documentation/stubs/qiskit.circuit.library.EvolvedOperatorAnsatz.html#qiskit.circuit.library.EvolvedOperatorAnsatz) is under N-local circuits.

### How can we reproduce the issue?

See the circuit library documentation: https://qiskit.org/documentation/apidoc/circuit_library.html

### What should happen?

[EvolvedOperatorAnsatz](https://qiskit.org/documentation/stubs/qiskit.circuit.library.EvolvedOperatorAnsatz.html#qiskit.circuit.library.EvolvedOperatorAnsatz) gate should be also in the N-local circuits directory https://qiskit.org/documentation/apidoc/circuit_library.html#n-local-circuits

### Any suggestions?

Move [EvolvedOperatorAnsatz](https://qiskit.org/documentation/stubs/qiskit.circuit.library.EvolvedOperatorAnsatz.html#qiskit.circuit.library.EvolvedOperatorAnsatz) gate to the N-local circuits directory https://qiskit.org/documentation/apidoc/circuit_library.html#n-local-circuits

### PR DESCRIPTION
**Pull request by Ca1eb3**
### Summary

Links to #8326 
Adds the EvolvedOperatorAnsatz class into the n_local directory which is its basis.
Comments on Original Pull request #10972 
Adds a deprecation warning to the EvolvedOperatorAnsatz class outside of the n_local directory as per this comment https://github.com/Qiskit/qiskit/pull/10972#issuecomment-1747199674 from the original pull request.

### Details and comments

The EvolvedOperatorAnsatz class which is not in the directory can now be deprecated in future releases.

### CODE REVIEW

File: test/python/circuit/library/test_evolved_op_ansatz.py
Comment (ShellyGarion) [inline test/python/circuit/library/test_evolved_op_ansatz.py:20]: this line should still work, right?

Comment (Ca1eb3) [reply test/python/circuit/library/test_evolved_op_ansatz.py:20]: It should I made the change so it imports EvolvedOperatorAnsatz the same way other n_local classes are imported to match the formatting more closely. Either way of importing should import the new class in n_local package.

File: qiskit/circuit/library/evolved_operator_ansatz.py
Comment (ShellyGarion) [inline qiskit/circuit/library/evolved_operator_ansatz.py]: Since both imports of `EvolvedOperatorAnsatz` from `qiskit.circuit.library` and `qiskit.circuit.library.n_local` should work, I'm not sure what should be the right message here.
@ElePT - could you please advise?

Comment (ElePT) [reply qiskit/circuit/library/evolved_operator_ansatz.py]: I believe that this is one of the edge cases of our deprecation policy. We normally have to follow the: `create new path-deprecate old path-remove old path` sequence. But in this case, the new path is included in the old one. Given that the `EvolvedOperatorAnsatz` is still importable from `qiskit.circuit.library`, users won't really feel a change, so I think the best course of action in this case is not to raise any deprecation warning and directly replace the old path with the new path (even if it might sound a bit contradictory). I do think it would be nice to keep a release note notifying the change.

File: qiskit/circuit/library/evolved_operator_ansatz.py
Comment (ElePT) [inline qiskit/circuit/library/evolved_operator_ansatz.py]: My suggestion here is to directly remove the old file (as I explained above, we are skipping to the last step of the process because this is an exception).

File: releasenotes/notes/move_evolved_operator_ansatz-5406bc071013c443.yaml
Comment (ElePT) [inline releasenotes/notes/move_evolved_operator_ansatz-5406bc071013c443.yaml]: As for the reno, I am actually leaning towards an `upgrades` entry notifying the move (it's not really a feature or a deprecation, but closer to a removal reno), with a text like:

> 
```
The class :class:`.EvolvedOperatorAnsatz` has been migrated from :mod:`qiskit.circuit.library` to :mod:`qiskit.circuit.library.n_local` to match the location of its base class. This move has not affected the usual import path of the class: ``from qiskit.circuit.library import EvolvedOperatorAnsatz``, but now it is also accessible doing ``from qiskit.circuit.library.n_local import EvolvedOperatorAnsatz``.
```

File: qiskit/circuit/library/evolved_operator_ansatz.py
Comment (Ca1eb3) [reply qiskit/circuit/library/evolved_operator_ansatz.py]: I made the changes to switch from the deprecation to the upgrade/immediate replacement.

File: qiskit/circuit/library/__init__.py
Comment (ShellyGarion) [inline qiskit/circuit/library/__init__.py:562]: Note that `EvolvedOperatorAnsatz` should also be moved from line 333 to the `N-local` circuits section, so it will appear in the correct place in the API docs.

**Review by ElePT — COMMENTED** [review]
Thanks for the fast response :). Besides Shelly's comment, the changes now look good to me (I will let her approve once this final docs detail is addressed).

File: qiskit/circuit/library/__init__.py
Comment (Ca1eb3) [reply qiskit/circuit/library/__init__.py:562]: Fixed.

**Review by ShellyGarion — APPROVED** [review]
LGTM, thank yo @Ca1eb3 for your contribution to Qiskit

**Comment by Ca1eb3** [comment]
@ShellyGarion @ElePT  Does this still need something from me? I know it has been approved but merging is still blocked and it still says a code owner review is required. Just making sure because this is my first PR to Qiskit.

**Review by ElePT — APPROVED** [review]

**Pull request merged**
